nntrainer/graph/graph_node.h

```diff
@@ -32,14 +32,16 @@ class GraphNode {
    * Each element indicates the orders with which the below operations
    * for each node are executed:
    * 1. Forwarding
-   * 2. calcGradient
-   * 3. calcDerivative
-   * 4. ApplyGradient
+   * 2. Recompute Forward (for gradient checkpointing)
```

Comment:
It is really great. However, we definitely need to check that every feature is working properly when we add this to the execution order. More importantly, we need to investigate the memory scheduling point more thoroughly.  

We can also compute the ideal peak memory when we enable this checkpointing through the memory scheduling, and we can also compare in an actual situation as well.